routing/localchans/manager.go

```diff
@@ -106,6 +110,17 @@ func (r *Manager) UpdatePolicy(newSchema routing.ChannelPolicy,
 		return err
 	}
 
+	// Return an error if any of the channels in chansToUpdate
+	// were not found.
+	if len(chansToUpdate) > 0 {
+		errStr := "can not find channel(s):"
+		for _, chanPoint := range chanPoints {
+			errStr += fmt.Sprintf(" %s:%d",
+				chanPoint.Hash.String(), chanPoint.Index)
+		}
+		return fmt.Errorf(errStr)
+	}
+
```

Comment:
would be cool to make the returned error a bit more informative. We can at this point tell if the channel is not found at all or if it is just unconfirmed or inactive. So can do something like this perhaps:

```
if len(chansToUpdate) > 0 {
		var errStr string
		for _, chanPoint := range chanPoints {
			channel, err := r.FetchChannel(chanPoint)
			if errors.Is(err, channeldb.ErrChannelNotFound) {
				errStr += fmt.Sprintf(
					"channel %s:%d not found; ",
					chanPoint.Hash.String(), chanPoint.Index,
				)
				continue
			}
			if err != nil {
				errStr += fmt.Sprintf(
					"error fetching channel %s:%d; ",
					chanPoint.Hash.String(), chanPoint.Index,
				)
				continue
			}
			if channel.IsPending {
				errStr += fmt.Sprintf("channel %s:%d is "+
					"not yet confirmed; ",
					chanPoint.Hash.String(), chanPoint.Index)
				continue
			}
			// ...
			// channel inactive perhaps?	
		}
		return fmt.Errorf(errStr)
	}
```

Reply:
We do not loop through any pending channels with `ForAllOutgoingChannels` so it isn't an issue at the moment.

Reply:
Great idea. I like it.